src/app/shared/pipes/truncate.pipe.ts

```diff
@@ -0,0 +1,13 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({
+  name: 'truncate',
+})
+export class TruncatePipe implements PipeTransform {
+  transform(value: string, args: any[]): string {
+    const limit = args.length > 0 ? parseInt(args[0], 10) : 20;
+    const trail = args.length > 1 ? args[1] : '...';
+    const limited = value.substring(0, limit);
+    return value.length > limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail : value;
+  }
```

Comment:
If we expect up to two arguments, we should make it explicit, including enforcing proper types:
```suggestion
  transform(value: string, [limit = 20, trail = '...']): string {
    const limited = value.substring(0, limit);
    return value.length > limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail : value;
  }
```
Although, why even provide an array if you can provide two positional parameters? Usage would be (not tested): `truncate:200:'...'`
```suggestion
  transform(value: string, limit = 20, trail = '...'): string {
    const limited = value.substring(0, limit);
    return value.length > limit ? limited.slice(0, limited.lastIndexOf(' ')) + trail : value;
  }
```